File: VB/GridViewBatchEdit/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports Models

Namespace GridViewBatchEdit.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
		End Function

		<ValidateInput(False)> _
		Public Function GridViewPartial() As ActionResult
			Return PartialView("_GridViewPartial", GridViewData.GridData)
		End Function
		<HttpPost, ValidateInput(False)> _
		Public Function DeleteRowPartial(ByVal ID As Integer) As ActionResult
			If ID >= 0 Then
				Try
					GridViewData.DeleteItem(ID)
				Catch e As Exception
					ViewData("EditError") = e.Message
				End Try
			End If
			Return PartialView("_GridViewPartial", GridViewData.GridData)
		End Function

	End Class
End Namespace

File: CS/GridViewBatchEdit/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using Models;

namespace GridViewBatchEdit.Controllers {
    public class HomeController: Controller {
        public ActionResult Index() {
            return View();
        }

        [ValidateInput(false)]
        public ActionResult GridViewPartial() {
            return PartialView("_GridViewPartial", GridViewData.GridData);
        }
        [HttpPost, ValidateInput(false)]
        public ActionResult DeleteRowPartial(int ID) {
            if (ID >= 0) {
                try {
                    GridViewData.DeleteItem(ID);
                }
                catch (Exception e) {
                    ViewData["EditError"] = e.Message;
                }
            }
            return PartialView("_GridViewPartial", GridViewData.GridData);
        }

    }
}